src/hiposfer/kamal/network/generators.clj

```diff
@@ -25,6 +25,9 @@
                      :node/location (network/->Location (lon-gen) (lat-gen))})]
     (concat nodes edges)))     ;; now connect the nodes
 
+;; TODO: this is far from ideal. Currently we just generate nodes here
+;; and then later on we generate ways and HOPE that they match. This
+;; can lead to bugs since the generated graph is not consistent
```

Comment:
Could you better explain why this inconsistency happen? Don't we use the same nodes for generating ways?

Reply:
The (possible) incosistency happens [here](https://github.com/hiposfer/kamal/pull/133/files#diff-ed89d5d21b7fca72e65792b7baddb867R53). It randomly picks nodes and creates an association between a way and several nodes.

The problem is that this association is subject to probability. Which means that there is a small chance that one or more nodes have no associated `way`. However for the OSM graph we assume that every node there has a way. Therefore the generated graph (might) contradict the assumptions of the real graph, which is definitely not what we want.

Reply:
Thanks for the info.